sources/engine/Stride.Graphics/OpenGL/CommandList.OpenGL.cs

```diff
@@ -920,39 +921,31 @@ public unsafe MappedResource MapSubresource(GraphicsResource resource, int subRe
                 mapMode = MapMode.WriteDiscard;
 
 
-            var buffer = resource as Buffer;
-            if (buffer != null)
+            if (resource is Buffer buffer)
             {
                 if (lengthInBytes == 0)
                     lengthInBytes = buffer.Description.SizeInBytes;
 
-                IntPtr mapResult = IntPtr.Zero;
-
                 GL.BindBuffer(buffer.BufferTarget, buffer.BufferId);
 
-#if !STRIDE_GRAPHICS_API_OPENGLES
-                //if (mapMode != MapMode.WriteDiscard && mapMode != MapMode.WriteNoOverwrite)
-                //    mapResult = GL.MapBuffer(buffer.bufferTarget, mapMode.ToOpenGL());
-                //else
-#endif
+                // Orphan the buffer (let driver knows we don't need it anymore)
+                if (mapMode == MapMode.WriteDiscard)
                 {
-                    // Orphan the buffer (let driver knows we don't need it anymore)
-                    if (mapMode == MapMode.WriteDiscard)
-                    {
-                        doNotWait = true;
-                        GL.BufferData(buffer.BufferTarget, (UIntPtr)buffer.Description.SizeInBytes, IntPtr.Zero, buffer.BufferUsageHint);
-                    }
+                    if(versionMajor >= 4 && versionMinor >= 3)
+                        GL.InvalidateBufferData(buffer.BufferId);
+                    else
+                        GL.BufferData(buffer.BufferTarget, (uint)buffer.Description.SizeInBytes, IntPtr.Zero, buffer.BufferUsageHint);
```

Comment:
My note here: Based on the various repositories I was inspired by, BufferData should work - Maybe the wrong size ? I don't know

Reply:
I'll add a reference to the documentation just in case https://registry.khronos.org/OpenGL-Refpages/gl4/html/glInvalidateBufferData.xhtml